### DISCUSSION TITLE
How to declare returning type when calling defineStore

### DISCUSSION THREAD

**Discussion by nickeljew** [vuejs/pinia#2518] [Help and Questions] [answered]
```
interface CounterStore {
  count: Ref<number>
}

const useCounterStore = defineStore<string, CounterStore>('counter', () => {
  const count = ref<number>(0)
  return { count }
})
```
Is it a proper way to declare returning type when calling defineStore?

**Comment by posva** [comment, marked as answer]
You don't, let Pinia infer the types for you. The only place you should put the types is at the level of the different properties:

```ts
defineStore('id', () => {
  const count = ref<number>(0) // <-- here
  return { count }
})
```

**Reply by nickeljew** [reply]
The inferred type is implicit and the type can’t be recognized by IDE.
I’m using TypeScript to explicitly declare types and let IDE recognize types of variables automatically.
Explicit typing helps to make codes clearer

**Reply by posva** [reply]
The code can be explicitly typed at the property level like in the example, and TypeScript is made to infer these types. You can notice how the code is way cleaner with just the `<number>` compared to the original example you posted with `<string, CounterStore>` and an extra interface **that needs to be updated every time you update the returned value** (or vice versa). If your IDE is not recognizing the types, there must be a misconfiguration somewhere as this is expected to work in TS. Make sure you have `strict: true` in your tsconfig
